cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunnerTest.java

```diff
@@ -185,6 +186,66 @@ public void testPushNoChanges() throws Exception {
     operationRunner.push(NAMESPACE, testRepoConfig, testAppDetails, testCommit);
   }
 
+  @Test
+  public void testListSuccess() throws Exception {
+    Path tmpRepoDirPath = TMP_FOLDER.newFolder().toPath();
+    Path file1 = tmpRepoDirPath.resolve("file1.json");
+    Path file2 = tmpRepoDirPath.resolve("file2.some.json");
+    Files.write(file1, new byte[]{});
+    Files.write(file2, new byte[]{});
+    // Skip non json file
+    Files.write(tmpRepoDirPath.resolve("file3"), new byte[]{});
+    // Skip directory
+    Files.createDirectories(tmpRepoDirPath.resolve("file4.json"));
+
+    Mockito.doReturn(FAKE_COMMIT_HASH).when(mockRepositoryManager).cloneRemote();
+    Mockito.doReturn(tmpRepoDirPath).when(mockRepositoryManager).getBasePath();
+    Mockito.doReturn("testHash1").when(mockRepositoryManager).getFileHash(file1.getFileName(), FAKE_COMMIT_HASH);
+    Mockito.doReturn("testHash2").when(mockRepositoryManager).getFileHash(file2.getFileName(), FAKE_COMMIT_HASH);
+    Mockito.doNothing().when(mockRepositoryManager).close();
+
+    List<ListAppResponse> listedApps = operationRunner.list(NAMESPACE, testRepoConfig).getApps();
+
+    Assert.assertEquals(2, listedApps.size());
+    Assert.assertEquals(listedApps.get(0).getName(), "file1");
+    Assert.assertEquals(listedApps.get(1).getName(), "file2.some");
+
+    Assert.assertEquals(listedApps.get(0).getFileHash(), "testHash1");
+    Assert.assertEquals(listedApps.get(1).getFileHash(), "testHash2");
```

Comment:
Nit: Since we have equals defined on ListAppResponse, we can use that for comparison.

Reply:
Done